operator/api/redpanda/v1alpha2/redpanda_types.go

```diff
@@ -97,62 +97,112 @@ type Migration struct {
 	ConsoleRef vectorizedv1alpha1.NamespaceNameRef `json:"consoleRef"`
 }
 
+// NodePoolStatus defines the observed state of any node pools tied to this cluster
+type NodePoolStatus struct {
+	// Name is the name of the pool
+	Name string `json:"name"`
+	// Replicas is the number of actual replicas currently across
+	// the node pool. This differs from DesiredReplicas during
+	// a scaling operation, but should be the same once the cluster
+	// has quiesced.
+	Replicas int32 `json:"replicas"`
+	// DesiredReplicas is the number of replicas that ought to be
+	// run for the cluster. It combines the desired replicas across
+	// all node pools.
+	DesiredReplicas int32 `json:"desiredReplicas"`
+	// OutOfDateReplicas is the number of replicas that don't currently
+	// match their node pool definitions. If OutOfDateReplicas is not 0
+	// it should mean that the operator will soon roll this many pods.
+	OutOfDateReplicas int32 `json:"outOfDateReplicas"`
+	// UpToDateReplicas is the number of replicas that currently match
+	// their node pool definitions.
+	UpToDateReplicas int32 `json:"upToDateReplicas"`
+	// CondemnedReplicas is the number of replicas that will be decommissioned
+	// as part of a scaling down operation.
+	CondemnedReplicas int32 `json:"condemnedReplicas"`
+	// ReadyReplicas is the number of replicas whose readiness probes are
+	// currently passing.
+	ReadyReplicas int32 `json:"readyReplicas"`
+	// RunningReplicas is the number of replicas that are actively in a running
+	// state.
+	RunningReplicas int32 `json:"runningReplicas"`
+}
+
 // RedpandaStatus defines the observed state of Redpanda
 type RedpandaStatus struct {
-	// Specifies the last observed generation.
-	// +optional
-	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
-
 	// Conditions holds the conditions for the Redpanda.
 	// +optional
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 
+	// LicenseStatus contains information about the current state of any
+	// installed license in the Redpanda cluster.
+	// +optional
+	LicenseStatus *RedpandaLicenseStatus `json:"license,omitempty"`
+
+	// NodePools contains information about the node pools associated
+	// with this cluster.
+	// +optional
+	NodePools []NodePoolStatus `json:"nodePools,omitempty"`
+
+	// everything below here is deprecated and should be removed
+
+	// Specifies the last observed generation.
+	// deprecated
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
```

Comment:
ObservedGeneration is pretty import IMO. Why are we deprecating it?

Reply:
So, in which case is this actually used? It seems like checking `ObservedGeneration` for `status.Type == "Quiesced"` or `status.Type == "Stable"` is pretty much what this is equivalent to -- the current `ObservedGeneration` that is set as a roll-up ensuring that all other status conditions have been updated.

Reply:
To be more specific -- `condition.Type == "Stable" && condition.Status == True && condition.ObservedGeneration == generation` seems to be what this actually is as it comes after everything succeeds without erroring: https://github.com/redpanda-data/redpanda-operator/blob/f66f19eb758827f3f1ed050136f1b4e8273998aa/operator/internal/controller/redpanda/redpanda_controller.go#L231